### PR TITLE
python3Packages.parse-type: 0.5.2 -> 0.5.6

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 0.5.6

- Cleanup

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 135646` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>python38Packages.behave</li>
    <li>python38Packages.cassandra-driver</li>
    <li>python38Packages.gremlinpython</li>
    <li>python38Packages.parse-type</li>
    <li>python38Packages.pytest-bdd</li>
    <li>python38Packages.python-docx</li>
    <li>python38Packages.radish-bdd</li>
    <li>python39Packages.behave</li>
    <li>python39Packages.cassandra-driver</li>
    <li>python39Packages.gremlinpython</li>
    <li>python39Packages.parse-type</li>
    <li>python39Packages.pytest-bdd</li>
    <li>python39Packages.python-docx</li>
    <li>python39Packages.radish-bdd</li>
    <li>terraform-compliance</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 135646` at be4a2559 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built successfully:</summary>
  <ul>
    <li>python38Packages.behave</li>
    <li>python38Packages.cassandra-driver</li>
    <li>python38Packages.gremlinpython</li>
    <li>python38Packages.parse-type</li>
    <li>python38Packages.pytest-bdd</li>
    <li>python38Packages.python-docx</li>
    <li>python38Packages.radish-bdd</li>
    <li>python39Packages.behave</li>
    <li>python39Packages.cassandra-driver</li>
    <li>python39Packages.gremlinpython</li>
    <li>python39Packages.parse-type</li>
    <li>python39Packages.pytest-bdd</li>
    <li>python39Packages.python-docx</li>
    <li>python39Packages.radish-bdd</li>
    <li>terraform-compliance</li>
  </ul>
</details>

File: pkgs/development/python-modules/parse-type/default.nix [hunk 1]
```diff
@@ -1,28 +1,48 @@
...
+    sha256 = "sha256-CJroqJIi5DpmR8i1lr8OJ+234615PhpVUsqK91XOT3E=";
   };
 
-  checkInputs = [ pytest pytest-runner ];
-  propagatedBuildInputs = [ parse six ] ++ lib.optional (pythonOlder "3.4") enum34;
+  propagatedBuildInputs = [
+    parse
+    six
+  ] ++ lib.optional (pythonOlder "3.4") [
+    enum34
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/parse-type/default.nix:26]:
```suggestion
```

File: pkgs/development/python-modules/parse-type/default.nix [hunk 2]
```diff
@@ -1,28 +1,48 @@
-{ lib, fetchPypi
-, buildPythonPackage, pythonOlder
-, pytest, pytest-runner
-, parse, six, enum34
+{ lib
+, buildPythonPackage
+, enum34
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/parse-type/default.nix:3]:
```suggestion
```

**Review by SuperSandro2000 — COMMENTED** [review]
We no longer support such old pythons.

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Pull request merged**